File: csharp/hamming/Hamming.cs
using System;

public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        int length = firstStrand.Length;
        if (secondStrand.Length != length) throw new ArgumentException();
        int count = 0;
        for (int i = 0; i < length; i++)
        {
            if (firstStrand[i] != secondStrand[i]) count++;
        }
        return count;
    }
}

File: fsharp/hamming/Hamming.fs
module Hamming

let distance (s1: string) (s2: string): int option = 
    if s1.Length <> s2.Length then None
    else Seq.zip s1 s2 |> Seq.map (fun (a, b) -> if a = b then 0 else 1) |> Seq.sum |> Some